### DISCUSSION TITLE
Redux devtools: example for actionSanitizer / stateSanitizer or deactivate entirely for production

### DISCUSSION THREAD

**Discussion by benomatis** [rematch/rematch#843] [Q&A] [answered]
Can I receive some help / an example on how to introduce actionSanitizer / stateSanitizer for Redux devtools in Rematch, or how I could maybe deactivate it completely for production?

**Comment by semoal** [comment, marked as answer]
I have to check actionSanitizer but you can disable reduxdevtools easily through Rematch init.
```js
import { init } from '@rematch/core'
export const store = init({
	models,
	redux: {
		devtoolOptions: {
			disabled: false,
		},
	},
})
```

**Reply by benomatis** [reply]
Strange, I never got a notification about this answer (only now that @monsterooo updated). Is that expected?

**Reply by semoal** [reply]
Inside redux->devtoolOptions you can pass any configuration accepted by redux-devtools-options, basically all this options:
https://github.com/zalmoxisus/redux-devtools-extension/blob/master/npm-package/index.d.ts

**Reply by benomatis** [reply]
I think this has been a misunderstanding: I was asking if it was expected that I would get no notification about your answer, because I didn't.

As to your answer, thank you, that appears to do it, will give it a go now!

**Reply by benomatis** [reply]
Working perfectly:

```js
const store = init({
    models,
    redux: {
        devtoolOptions: {
            disabled: process.env.NODE_ENV === 'production'
        }
    }
})
```

Thank you!

**Reply by semoal** [reply]
> I think this has been a misunderstanding: I was asking if it was expected that I would get no notification about your answer, because I didn't.
> 
> As to your answer, thank you, that appears to do it, will give it a go now!

Actually discussions are in beta so probably sometimes isn't working properly

**Comment by monsterooo** [comment]
[Extension crashes while the reducer is working](https://github.com/zalmoxisus/redux-devtools-extension/issues/530)  actionSanitizer / stateSanitizer Very helpful at some point 🙏

**Reply by semoal** [reply]
```ts
export const store = init<RootModel, FullModel>({
	redux: {
		devtoolOptions: {
			stateSanitizer: <S extends Partial<RootState>>(state: S): S => {
				const { cart, ...rest } = state

				return {
					...rest,
				} as S
			},
		},
	},
```
You can use it also on Typescript, i'm going to push with the actual types from the official redux-devtools project. 

For me is working properly, if you find any bug feel free to create an issue
